msw/2/utils.ts

```diff
@@ -0,0 +1,273 @@
+import {
+	type BindingElement,
+	type Block,
+	type PropertyAccessExpression,
+	type SourceFile,
+	SyntaxKind,
+	type CallExpression,
+	type FunctionExpression,
+	type ArrowFunction,
+	ParameterDeclaration,
+} from 'ts-morph';
+
+// It would be nice to share these instead of duplicating the code.
+
+export function shouldProcessFile(sourceFile: SourceFile) {
+	return !!sourceFile
+		.getImportDeclarations()
+		.find((decl) =>
+			decl.getModuleSpecifier().getLiteralText().startsWith('msw'),
+		);
+}
+
+export function addNamedImportDeclaration(
+	sourceFile: SourceFile,
+	moduleSpecifier: string,
+	name: string,
+) {
+	const importDeclaration =
+		sourceFile.getImportDeclaration(moduleSpecifier) ||
+		sourceFile.addImportDeclaration({ moduleSpecifier });
+
+	if (
+		importDeclaration
+			.getNamedImports()
+			.some((specifier) => specifier.getName() === name)
+	) {
+		return importDeclaration;
+	}
+
+	return importDeclaration.addNamedImport({ name });
+}
+
+export function getImportDeclarationAlias(
+	sourceFile: SourceFile,
+	moduleSpecifier: string,
+	name: string,
+) {
+	const importDeclaration = sourceFile.getImportDeclaration(moduleSpecifier);
+	if (!importDeclaration) {
+		return null;
+	}
+
+	const namedImport = importDeclaration
+		.getNamedImports()
+		.find((specifier) => specifier.getName() === name);
+
+	if (!namedImport) {
+		return null;
+	}
+
+	return namedImport.getAliasNode()?.getText() || namedImport.getName();
+}
+
+export function searchIdentifiers(
+	codeBlock: Block,
+	searchables: string[],
+	matchCb?: (matchedVal: string) => void,
+) {
+	const matched = new Set<string>();
+	for (const parent of (
```

Comment:
oh, here you can do the same trick with `[...a, ...b]` concatenation that I showed before.